homeassistant/components/sensor/metoffice.py

```diff
@@ -68,63 +70,70 @@
 }
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_NAME, default=None): cv.string,
     vol.Required(CONF_API_KEY): cv.string,
     vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
         vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Inclusive(CONF_LATITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.latitude,
+    vol.Inclusive(CONF_LONGITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.longitude,
 })
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
-    """Set up the Metoffice sensor platform."""
+    """Set up the Met Office sensor platform."""
     import datapoint as dp
-    datapoint = dp.connection(api_key=config.get(CONF_API_KEY))
-
+    api_key = config.get(CONF_API_KEY)
     latitude = config.get(CONF_LATITUDE, hass.config.latitude)
     longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    name = config.get(CONF_NAME)
+
+    datapoint = dp.connection(api_key=api_key)
 
     if None in (latitude, longitude):
         _LOGGER.error("Latitude or longitude not set in Home Assistant config")
-        return False
+        return
 
     try:
-        site = datapoint.get_nearest_site(latitude=latitude,
-                                          longitude=longitude)
+        site = datapoint.get_nearest_site(
+            latitude=latitude, longitude=longitude)
     except dp.exceptions.APIException as err:
         _LOGGER.error("Received error from Met Office Datapoint: %s", err)
-        return False
+        return
 
     if not site:
         _LOGGER.error("Unable to get nearest Met Office forecast site")
-        return False
+        return
 
-    # Get data
     data = MetOfficeCurrentData(hass, datapoint, site)
     try:
         data.update()
     except (ValueError, dp.exceptions.APIException) as err:
```

Comment:
I don't think we should reraise the error in `data.update`. It's already logged there and raising it again will add an ugly stack trace to the log of it happens after setup_platform is finished.

Instead we can check if `data.data` is `None`, which it shouldn't be if everything is successful.

Reply:
We can do that.